### PR TITLE
Implement download image

### PR DESCRIPTION
**Pull request by Themanwhosmellslikesugar**

### CODE REVIEW

File: src/modules/Dashboard/components/CardImage/CardImage.jsx [hunk 1]
```diff
@@ -82,7 +84,11 @@ const CardImage = ({ deleteCurrentImageAction, image }) => {
                 <Card.Footer>
                     <Row>
                         <Col xs={6}>
-                            <Button variant="primary">
+                            <Button
+                                variant="primary"
+                                disabled={image.status !== "Succeeded"}
+                                href={`${downloadImageUrl}/${image.image_id}.img`}
```
Comment (karpa4o4) [inline src/modules/Dashboard/components/CardImage/CardImage.jsx:90]: Put the file extension in variable too.

File: src/modules/Dashboard/components/CardImage/CardImage.jsx [hunk 2]
```diff
@@ -84,7 +99,11 @@ const CardImage = ({ deleteCurrentImageAction, image }) => {
                 <Card.Footer>
                     <Row>
                         <Col xs={6}>
-                            <Button variant="primary">
+                            <Button
```
Comment (karpa4o4) [inline src/modules/Dashboard/components/CardImage/CardImage.jsx:102]: Add the `download` attribute to `a` tag. See [here](https://www.w3schools.com/howto/howto_html_download_link.asp)

File: src/modules/Dashboard/components/CardImage/CardImage.jsx [hunk 3]
```diff
@@ -84,7 +99,11 @@ const CardImage = ({ deleteCurrentImageAction, image }) => {
...
+                                variant="primary"
```
Comment (karpa4o4) [inline src/modules/Dashboard/components/CardImage/CardImage.jsx:103]: Remove `variant` prop, `primary` option is the default.

File: src/modules/Dashboard/components/DownloadInfoModal/DownloadInfoModal.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+const statusText = {
+    Undefined: failedText,
+    Failed: failedText,
+    Interrupted: failedText,
+    Pending: waitText,
+    Building: waitText,
+};
+
+const DownloadInfoModal = ({
```
Comment (karpa4o4) [inline src/modules/Dashboard/components/DownloadInfoModal/DownloadInfoModal.jsx:16]: You can place props on the same line.

File: src/modules/Dashboard/components/DownloadInfoModal/DownloadInfoModal.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+import React from "react";
+import { Modal } from "react-bootstrap";
+
+const failedText = "Build image is failed, try build it again";
+const waitText = "Your image is currently building, wait until the build is complete";
```
Comment (eugulixes) [inline src/modules/Dashboard/components/DownloadInfoModal/DownloadInfoModal.jsx:6]: Your image is currently being built. Wait until the build is done.

File: src/modules/Dashboard/components/DownloadInfoModal/DownloadInfoModal.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+import React from "react";
+import { Modal } from "react-bootstrap";
+
+const failedText = "Build image is failed, try build it again";
```
Comment (eugulixes) [inline src/modules/Dashboard/components/DownloadInfoModal/DownloadInfoModal.jsx:5]: The build is failed. Try to build it again.

**Review by eugulixes — APPROVED** [review]

**Review by karpa4o4 — APPROVED** [review]

**Pull request merged**